text/0000-plugin-package-format.md

```diff
@@ -0,0 +1,111 @@
+
+#### `windows`
+The Windows platform supports the additional keywords `<version>`, `<arch>` and `<bits>`, which should be written as: 
+
+```
+/<name>/bin/windows[/<version>]/<arch>-<bits>/<name>[.<ext>]
+```
+
+- `<version>` is optional.
+- `<arch>` is required.
+- `<bits>` is required.
+
+This layout allows us to (optionally) specify a minimum platform version, a required architecture, and a required bitness. For example, with this we can filter for Windows 10 20H2 x86 64-bits with the following text: `/<name>/bin/windows/10.0.19042/x86-64/<name>[.<ext>]`, and as such prevent the plugin from loading on unsupported platforms automatically.
+
+#### `mac` or `macos`
+The Mac/MacOS platform supports the additional keywords `<version>`, `<arch>` and `<bits>`, which should be written as:
+
+```
+/<name>/bin/mac[/<version>]/<arch>-<bits>/<name>[.<ext>]
+```
+
+- `<version>` is optional.
+- `<arch>` is required.
+- `<bits>` is required.
+
+This layout allows us to (optionally) specify a minimum platform version, an required architecture, and a required bitness. We can easily filter for specific Apple devices, and require certain OS versions for plugins, or even have M1 or X86 exclusive plugins.
+
+#### `linux`
+The Linux platform supports the additional keywords `<distro>`, `<version>` `<arch>` and `<bits>`, which should be written as:
+
+```
+/<name>/bin/linux[/<distro>[-<version>]]/<arch>-<bits>/<name>[.<ext>]
+```
+
+- `<distro>` is optional, and specific distributions may have a fallback to other values if no specific version is present.
+- `<version>` is optional.
+- `<arch>` is required.
+- `<bits>` is required.
+
+### Keyword: `<version>`
+Specifies a minimum version in the format `A[.B[.C[.D]]]`, where each component is a positive integer number. Each platform can parse this information how they want, and may specify if this keyword is optional or not.
+
+### Keyword: `<distro>`
+Linux exclusive keyword to specify the exact distribution on which the plugin can run, such as `ubuntu`, `debian`, `freebsd`, etc. If a specific distro is missing, a fallback may be used to load binaries from another distro known to be compatible (for example, `ubuntu` may load `debian` and vice versa).
+
+### Keyword: `<arch>`
+The architecture keyword is used to denote the architecture, to noones surprise. These are valid:
+
+- `any`
+    Runs on any architecture, see FatELF on Linux or Universal Binary on MacOS
+- `x86` or `X86`
+    Intel/AMD x86 based system.
+- `arm` or `ARM`
+    ARM based system, such as a Windows ARM laptop, or a Apple MacBook Pro with M1.
+
+### Keyword: `<bits>`
+Finally, the bitness of the system. This is an integer of any value, but the common numbers are:
+
+- `any`: Any bitness, see FatELF on Linux or Universal Binary on MacOS.
+- `32`: 32-bits
+- `64`: 64-bits
+
+Future bitness support is easily added by simply just increasing the number, so if we ever end up with 128-bits, just use 128.
+
+## Behavior when loading Plugins
```

Comment:
I vote this to be extended a bit:

1. All built-in plugins should be loaded before anything else
2. Plugins in the new format should be loaded second, as they'll be in unique directories and therefore their structure/loading should be predictable
3. Plugins in the original/legacy format should be loaded last, and if any of them have a name matching a built-in or new-style plugin, should (by default) **not** be enabled (though a UI prompt somewhere to allow loading duplicate legacy plugins could be considered)

Reply:
That sounds better than the current text, I'll adjust it.

Reply:
Adjusted.